File: shared/src/main/java/net/ncguy/ability/Requirement.java
package net.ncguy.ability;

import net.ncguy.classes.Classes;

import java.util.ArrayList;

public interface Requirement {

    class ClassRequirement extends ArrayList<Classes> implements Requirement {}
    class LevelRequirement implements Requirement {

        public LevelRequirement() { }

        public LevelRequirement(int level) {
            this.level = level;
        }

        public int level;
    }

}

File: Tools/Tools/Shared/Data/Requirement.cs
using System;
using System.Collections.Generic;

namespace Shared.Data {
    public abstract class Requirement {

    }

    public class ClassRequirement : Requirement {
        public List<Classes> Classes = new List<Classes>();
    }

    public class LevelRequirement : Requirement {
        public int Level;
    }

}